lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java

```diff
@@ -103,10 +153,17 @@ public static int binarySearch(int[] values, int start, int end, int value) {
 
     // TODO(you): Finish implementing this algorithm
 
+    if (values[pivotIndex] == value) {
+      return pivotIndex;
+    } else if (values[pivotIndex] > value) {
+      return binarySearch(values, start, pivotIndex - 1, value);
```

Comment:
If you're up for it, here's another challenge: implement this without recursion (aka the iterative approach).

Reply:
This was tricky! definitely helping wrap my head around the concepts though.